model/patch/patch.go

```diff
@@ -301,6 +302,20 @@ func (p *Patch) SetParameters(parameters []Parameter) error {
 	)
 }
 
+func (p *Patch) SetDownstreamParameters(parameters []Parameter) error {
+	for _, param := range parameters {
+		p.Triggers.DownstreamParameters = append(p.Triggers.DownstreamParameters, param)
```

Comment:
in the case where a parameter already exists the parameter list the in-memory patch and db patch will diverge. I would think to either: 1) check here a parameter doesn't already exist in the slice before appending it or 2) use `$push` instead of `$addToSet`